tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -348,15 +348,21 @@ public async Task ConcurrentJobConfigurationShouldBeRespected()
             .And.WithCronExpression("* * * * *").WithName("Job 2")
             .And.WithCronExpression("* * * * *").WithName("Job 3")
             .And.WithCronExpression("* * * * *").WithName("Job 4")));
+
         var provider = CreateServiceProvider();
 
+        (IDisposable subscriber, IList<ExecutionProgress> events) = RegisterAnExecutionProgressSubscriber(provider);
+
         await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
 
         FakeTimer.Advance(TimeSpan.FromMinutes(1));
-        // Wait 2 instances at the same time
-        (await WaitForJobsOrTimeout(2, TimeSpan.FromMilliseconds(150))).ShouldBeTrue();
-        // But not another instance
-        (await WaitForJobsOrTimeout(1, TimeSpan.FromMilliseconds(50))).ShouldBeFalse();
+        await Task.Delay(TimeSpan.FromMilliseconds(1), CancellationToken);
+
+        subscriber.Dispose();
+
+        var runningJobs = events.Where(e => e.State == ExecutionState.Initializing).ToList();
+        Assert.Equal(2, runningJobs.Count);
+        Assert.NotEqual(runningJobs[0].CorrelationId, runningJobs[1].CorrelationId);
```

Comment:
@linkdotnet Are those expectations similar to what was previously asserted?

Reply:
Yes - it was all about that we have the same job (that can run two instances in parallel) is also scheduled and triggered accordingly. That seems to be correct